### PR TITLE
bpo-36052: Raise a SyntaxError when we assign a value to __debug__

### PR DESCRIPTION
**Pull request by matrixise**
Co-authored-by: Stéphane Wirtel <stephane@wirtel.be>
Co-authored-by: Pablo Galindo <Pablogsal@gmail.com>

https://bugs.python.org/issue36052

### CODE REVIEW

File: Lib/test/test_syntax.py [hunk 1]
```diff
@@ -626,6 +626,9 @@
     Traceback (most recent call last):
     SyntaxError: cannot assign to __debug__
 
+>>> (__debug__ := 'spam')
```
Comment (pablogsal) [inline Lib/test/test_syntax.py]: Please, move this one after the check for `>>> __debug__ = 1` :+1: 
https://github.com/python/cpython/blob/a5f32adf1db51799f37009b82747dc8e6fdfefa4/Lib/test/test_syntax.py#L50

**Review by pablogsal — CHANGES_REQUESTED** [review]

File: Lib/test/test_syntax.py (hunk 1 above)
Comment (matrixise) [reply Lib/test/test_syntax.py]: done, and I have added you as Co-authored-by

File: Lib/test/test_syntax.py [hunk 2]
```diff
@@ -625,7 +629,6 @@
     >>> def f(*xx, __debug__): pass
     Traceback (most recent call last):
     SyntaxError: cannot assign to __debug__
-
```
Comment (pablogsal) [inline Lib/test/test_syntax.py]: Sorry for the nit: Could you restore this empty line?

**Review by pablogsal — APPROVED** [review]
LGTM!

Thanks for fixing this one @matrixise  :)

File: Misc/NEWS.d/next/Core and Builtins/2019-02-20-17-57-31.bpo-36052.l8lJSi.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Raise a SyntaxError when we try to assign a value to `__debug__` with the
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2019-02-20-17-57-31.bpo-36052.l8lJSi.rst]: Also, can you add markup syntax to ``SyntaxError``?

File: Lib/test/test_syntax.py (hunk 2 above)
Comment (matrixise) [reply Lib/test/test_syntax.py]: done

File: Misc/NEWS.d/next/Core and Builtins/2019-02-20-17-57-31.bpo-36052.l8lJSi.rst (hunk 3 above)
Comment (matrixise) [reply Misc/NEWS.d/next/Core and Builtins/2019-02-20-17-57-31.bpo-36052.l8lJSi.rst]: done

**Comment by matrixise** [comment]
sorry, I have to leave and will continue the fix (if there is a fix to do ;-)). See you later (+- 3hours)

File: Misc/NEWS.d/next/Core and Builtins/2019-02-20-17-57-31.bpo-36052.l8lJSi.rst (hunk 3 above)
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2019-02-20-17-57-31.bpo-36052.l8lJSi.rst]: I thin we don't use 'we' in the docs. Maybe:

 ```
Raise a ``SyntaxError`` when assigning a value to `__debug__` using the assignment operator.
```

**Review by serhiy-storchaka — APPROVED** [review]

File: Misc/NEWS.d/next/Core and Builtins/2019-02-20-17-57-31.bpo-36052.l8lJSi.rst (hunk 3 above)
Comment (matrixise) [reply Misc/NEWS.d/next/Core and Builtins/2019-02-20-17-57-31.bpo-36052.l8lJSi.rst]: Updated

**Comment by matrixise** [comment]
@pablogsal or @serhiy-storchaka who want to merge this PR? ;-)

**Pull request merged**